apps/src/pythonlab/PythonlabView.tsx

```diff
@@ -1,14 +1,120 @@
 import moduleStyles from './pythonlab-view.module.scss';
-import SidePanel from './SidePanel';
+import {ConfigType} from '@cdo/apps/weblab2/CDOIDE/types';
+import Editor from '../weblab2/CDOIDE/CenterPane/Editor';
+import {LanguageSupport} from '@codemirror/language';
+import {python} from '@codemirror/lang-python';
+import {CDOIDE} from '../weblab2/CDOIDE';
+import {MultiFileSource} from '../lab2/types';
+import Lab2Registry from '../lab2/Lab2Registry';
+import {useAppDispatch, useAppSelector} from '../util/reduxHooks';
+import {setSource} from './pythonlabRedux';
+import PythonConsole from './PythonConsole';
+
+const pythonlabLangMapping: {[key: string]: LanguageSupport} = {
+  py: python(),
+};
+
+const defaultProject: MultiFileSource = {
+  files: {
+    '0': {
+      id: '0',
+      name: 'main.py',
+      language: 'py',
+      contents: 'print("Hello world!")',
+      folderId: '1',
+      active: true,
+      open: true,
+    },
+  },
+  folders: {
+    '1': {
+      id: '1',
+      name: 'src',
+      parentId: '0',
+    },
+  },
+};
+
+const defaultConfig: ConfigType = {
+  //showSideBar: true,
+  // showLeftNav: false,
+  // showEditor: false,
+  showPreview: false,
+  activeLeftNav: 'Files',
+  EditorComponent: () => Editor(pythonlabLangMapping, ['py']),
+  leftNav: [
+    {
+      icon: 'fa-square-check',
+      component: 'Instructions',
+    },
+    {
+      icon: 'fa-file',
+      component: 'Files',
+    },
+    {
+      icon: 'fa-solid fa-magnifying-glass',
+      component: 'Search',
+    },
+  ],
+  sideBar: [
+    {
+      icon: 'fa-circle-question',
+      label: 'Help',
+      action: () => window.alert('Help is not currently implemented'),
+    },
+    {
+      icon: 'fa-folder',
+      label: 'Files',
+      action: () => window.alert('You are already on the file browser'),
+    },
+  ],
+  instructions: 'Welcome to Python Lab!',
+};
 
 const PythonlabView: React.FunctionComponent = () => {
+  const [currentProject, setCurrentProject] = useState<MultiFileSource>();
+  const [config, setConfig] = useState<ConfigType>(defaultConfig);
+  const initialSources = useAppSelector(state => state.lab.initialSources);
+  const channelId = useAppSelector(state => state.lab.channel?.id);
+  const dispatch = useAppDispatch();
+
+  // TODO: This is repeated in Weblab2View. Can we extract this out somewhere?
+  const setProject = (newProject: MultiFileSource) => {
+    setCurrentProject(newProject);
+    dispatch(setSource(newProject));
+    if (Lab2Registry.getInstance().getProjectManager()) {
+      const projectSources = {
+        source: newProject,
+      };
+      Lab2Registry.getInstance().getProjectManager()?.save(projectSources);
+    }
+  };
+
+  useEffect(() => {
+    // We reset the project when the channelId changes, as this means we are on a new level.
+    setCurrentProject(
+      (initialSources?.source as MultiFileSource) || defaultProject
+    );
+  }, [channelId, initialSources]);
+
```

Comment:
Not for this PR (unless you want to), but I think yoinking it out into a hook that we can share with weblab sounds good.

Something like this:

```
// or whatever we call the hook
export const useIDEProject = () => {
  const [currentProject, setCurrentProject] = useState<MultiFileSource>();
  const initialSources = useAppSelector(state => state.lab.initialSources);
  const channelId = useAppSelector(state => state.lab.channel?.id);
  const dispatch = useAppDispatch();

  // I think I memoized this correctly...
  const setProject = useMemo(() => (newProject: MultiFileSource) => {
    setCurrentProject(newProject);
    dispatch(setSource(newProject));
    if (Lab2Registry.getInstance().getProjectManager()) {
      const projectSources = {
        source: newProject,
      };
      Lab2Registry.getInstance().getProjectManager()?.save(projectSources);
    }
  }, [setCurrentProject, dispatch])

  useEffect(() => {
    // We reset the project when the channelId changes, as this means we are on a new level.
    setCurrentProject(
      (initialSources?.source as MultiFileSource) || defaultProject
    );
  }, [channelId, initialSources]);

  return [currentProject, setProject]
}
```

I think that'll do it, but double check my work. Then in both components we can just do:

```
const [project, setProject] = useIDEProject()
```

and then it should handle all of the IO and updates for us and keep it nicely hidden.

It might be useful to have another wrapper component that automatically sets these `project`/`setProject` values on the IDE, but that's kinda feeling like a premature optimization atm. If there's a way to add it w/o another wrapper, let's go for it, but otherwise maybe wait a little.

Reply:
I made a follow up task to not lose this! https://codedotorg.atlassian.net/browse/CT-499